libcloud/compute/drivers/onapp.py

```diff
@@ -0,0 +1,842 @@
+                                this network interface
+        :type  network_join_id: ``int``
+
+        :param rate_limit: set the port speed of a network interface you wish
+                           to attach
+        :type  rate_limit: ``int``
+
+        :param primary: set True if the interface is primary. Otherwise False.
+        :type  primary: ``bool``
+        """
+        data = {"network_interface": {
+            "label": label,
+            "network_join_id": network_join_id,
+            "primary": primary,
+            "rate_limit": rate_limit
+        }}
+
+        response = self.connection.request(
+            "/virtual_machines/%s/network_interfaces.json" % identifier,
+            data=json.dumps(data),
+            headers={
+                "Content-type": "application/json"},
+            method="POST")
+
+        return OnAppNetworkInterface(**response.object["network_interface"])
+
+    def delete_network_interface(self, identifier, network_interface_id):
+        """
+        Delete a network interface from virtual machine
+
+        :param identifier: VS identifier
+        :type  identifier: ``str``
+
+        :param network_interface_id: Network interface id
+        :type  identifier: ``int``
+        """
+        action = "/virtual_machines/{identifier}/network_interfaces/{network_interface_id}.json".format(
+            identifier=identifier, network_interface_id=network_interface_id)
+
+        self.connection.request(action, method="DELETE")
+
+    def edit_network_interface(self, identifier, network_interface_id,
+                               label=None, primary=None, rate_limit=None):
+        """
+        :param identifier: VS identifier
+        :type  identifier: ``str``
+
+        :param network_interface_id: Network interface id
+        :type  identifier: ``int``
+
+        :param label: network interfaces label
+        :type  label: ``str``
+
+        :param primary: 1 if this network interface is primary, otherwise 0
+        :type  primary: ``bool``
+
+        :param rate_limit: port speed in Mbps
+        :type  rate_limit: ``int``
+        :return:
+        """
+        data = {"network_interface": {
+            "rate_limit": rate_limit
+        }}
+        if label is not None:
+            data["network_interface"]["label"] = label
+        if primary is not None:
+            data["network_interface"]["primary"] = primary
+
+        action = "/virtual_machines/{identifier}/network_interfaces/{network_interface_id}.json".format(
+            identifier=identifier, network_interface_id=network_interface_id)
+
+        self.connection.request(action,
+                                data=json.dumps(data),
+                                headers={
+                                    "Content-type": "application/json"},
+                                method="PUT")
+
+    #
+    # Methods to be implemented
+    #
+    def list_sizes(self, location=None):
+        pass
```

Comment:
Instead of having those methods here, either add `raise NotImplementedError` or remove them - base NodeDriver class already throws this error by default if the method is not overridden in a sub class.

This way it's possible to programatically check and know that those methods are unimplemented. Python returns implicit `None` by default so doing that means it's hard / impossible to programatically check if those methods are implemented.